rero_ils/modules/documents/dumpers/indexer.py

```diff
@@ -164,28 +165,45 @@ def _process_identifiers(record, data):
             ):
                 data[key] = filtered_identifiers
 
-    def dump(self, record, data):
-        """Dump a document instance with basic document informations.
+    @staticmethod
+    def _process_i18n_entities(record, data):
+        """Process fields containing entities to allow i18n search."""
+        # Contribution (aka. authors of the document)
+        if contributions := data.pop('contribution', []):
+            data['contribution'] = process_i18n_literal_fields(contributions)
+        # Subject (could contain subdivisions to perform too)
+        if subjects := data.pop('subjects', []):
+            data['subjects'] = process_i18n_literal_fields(subjects)
+        if genreForms := data.pop('genreForm', []):
+            data['genreForm'] = process_i18n_literal_fields(genreForms)
 
-        :param record: The record to dump.
-        :param data: The initial dump data passed in by ``record.dumps()``.
-        """
-        from rero_ils.modules.local_fields.api import LocalField
+    @staticmethod
+    def _process_sort_title(record, data):
+        """Compute and store the document title used to sort it."""
         from rero_ils.utils import language_mapping
+        sort_title = TitleExtension.format_text(data.get('title', []))
+        language = language_mapping(data.get('language')[0].get('value'))
+        if current_app.config.get('RERO_ILS_STOP_WORDS_ACTIVATE', False):
+            sort_title = current_app. \
+                extensions['reroils-normalizer-stop-words']. \
+                normalize(sort_title, language)
+        data['sort_title'] = sort_title
 
-        from ..api import Document
-        self._process_holdings(record, data)
-
-        if contributions := process_literal_contributions(
-            data.get('contribution', [])
-        ):
-            data.pop('contribution', None)
-            data['contribution'] = contributions
-
-        # TODO: compare record with those in DB to check which authors have
-        # to be deleted from index
+    @staticmethod
+    def _process_local_field(record, data):
+        """Add local field data related to this document."""
+        from rero_ils.modules.local_fields.api import LocalField
+        fields = list(LocalField.get_local_fields_by_id('doc', record['pid']))
```

Comment:
because if generator return empty dataset, it will cause an empty array into ES document ... not really incorrect, but in the project, for such case, we don't sent any data instead of empty array (there is a uni-test about that)